### ISSUE TITLE
Disabling point compression in proving key serialization

### ISSUE THREAD

**Issue by ebfull** [scipr-lab/libsnark#28]
Large constraint systems mean huge proving keys, and with point compression enabled, loading those keys takes several minutes. Right now you can only enable point compression across the board, and it could be tricky to refactor that. Perhaps it should be made into a global that can be selectively disabled/enabled by the end user?

**Comment by daira** [comment]
Specifically, I think we need to be able to enable compression for proofs, but disable it for the proving key.

**Comment by tromer** [comment]
Makes sense. But de/compression happens deep inside the context-independent de/serialization code. Controlling it via a global that changes the interpretation of all streams is pretty scary.

C++ streams supports a notion of flags (`setf()` etc.), which would have been nice, but it's not extendible.

**Comment by daira** [comment]
The [de]serialisation of proofs doesn't necessarily have to be done using the stream operators.
